src/ptf/testutils.py

```diff
@@ -2396,5 +2431,27 @@ def count_matched_packets(test, exp_packet, port, device_number=0, timeout=1):
 
     return total_rcv_pkt_cnt
 
+def count_matched_packets_all_ports(test, exp_packet, ports=[], device_number=0, timeout=1):
+    """
+    Receive all packets on all specified ports and count how many expected packets were received.
+    As soon as the packets stop arriving, the function waits for the timeout value and returns the
+    cumulative counter
+    """
+    last_matched_packet_time = time.time()
+    total_rcv_pkt_cnt = 0
+    while True:
+        if (timeout >= 0) and ((time.time() - last_matched_packet_time) > timeout):
+            break
+
+        (rcv_device, rcv_port, rcv_pkt, pkt_time) = dp_poll(test, device_number=device_number, timeout=timeout)
```

Comment:
My thought in this design is that a 'timeout' occurs once we stop receiving expected packets for the specified duration. This could occur one of two ways: 1.) dp_poll times out (meaning packets completely stopped arrived for the timeout duration, or 2.) dp_poll continues receiving packets, but we haven't seen an expected packet for the timeout duration.